interfaces/builtin/rtc.go

```diff
@@ -0,0 +1,115 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2016 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package builtin
+
+import (
+	"bytes"
+	"fmt"
+
+	"github.com/snapcore/snapd/interfaces"
+)
+
+const rtcControlConnectedPlugAppArmor = `
+# Description: Allow write access to system real-time clock
+# See 'man 4 rtc' for details.
+
+/dev/rtc rw,
+/dev/rtc[0-9]* rw,
+
+# As the core snap ships the hwclock utility we can also allow
+# clients to use it now that they have access to the relevant
+# device nodes.
+/sbin/hwclock ixr,
+`
+
+// The type for the rtc interface
+type RtcInterface struct{}
```

Comment:
AFAIK we use the 'rtc' variant for all things where no direct interaction is covered by the interface like 'iio', 'tpm', or 'i2c'. Add -control as suffix would mean the interface gives some privileged access to a resource and -observe would give just read-only access. That is so far a common scheme in the interface naming and was also expressed by @niemeyer and @jdstrand as the way to go.